airbyte-integrations/bases/base-normalization/normalization/transform_catalog/utils.py

```diff
@@ -17,100 +17,88 @@ def remove_jinja(command: str) -> str:
     return str(command).replace("{{ ", "").replace(" }}", "")
 
 
-# transform oneOf nested list ot plain types list
-def get_plain_list_from_one_of_array(one_of_definition) -> list:
-    one_of_property = one_of_definition[data_type.ONE_OF_VAR_NAME]
-    plain_datatypes_list = list()
-    for prop in one_of_property:
-        plain_datatypes_list.append(prop[data_type.REF_TYPE_VAR_NAME])
-    return plain_datatypes_list
-
-
-def is_string(property_type) -> bool:
-    if data_type.ONE_OF_VAR_NAME in property_type and data_type.ONE_OF_VAR_NAME in property_type:
-        plain_datatypes_list = get_plain_list_from_one_of_array(property_type)
-        return plain_datatypes_list == data_type.STRING_TYPE or data_type.STRING_TYPE in plain_datatypes_list
+def is_string(definition: dict) -> bool:
+    if data_type.ONE_OF_VAR_NAME in definition:
+        return bool(any(option[data_type.REF_TYPE_VAR_NAME] == data_type.STRING_TYPE for option in definition[data_type.ONE_OF_VAR_NAME]))
     else:
-        return property_type == data_type.STRING_TYPE or data_type.STRING_TYPE in property_type
+        return data_type.REF_TYPE_VAR_NAME in definition and data_type.STRING_TYPE == definition[data_type.REF_TYPE_VAR_NAME]
 
 
-def is_binary_datatype(property_type) -> bool:
-    if data_type.ONE_OF_VAR_NAME in property_type and data_type.ONE_OF_VAR_NAME in property_type:
-        plain_datatypes_list = get_plain_list_from_one_of_array(property_type)
-        return plain_datatypes_list == data_type.BINARY_DATA_TYPE or data_type.BINARY_DATA_TYPE in plain_datatypes_list
-    else:
-        return property_type == data_type.BINARY_DATA_TYPE or data_type.BINARY_DATA_TYPE in property_type
+def is_binary_datatype(definition: dict) -> bool:
+    return data_type.REF_TYPE_VAR_NAME in definition and data_type.BINARY_DATA_TYPE == definition[data_type.REF_TYPE_VAR_NAME]
 
 
 def is_datetime(definition: dict) -> bool:
     return is_datetime_with_timezone(definition) or is_datetime_without_timezone(definition)
 
 
 def is_datetime_without_timezone(definition: dict) -> bool:
-    if data_type.ONE_OF_VAR_NAME in definition and data_type.ONE_OF_VAR_NAME in definition:
-        plain_datatypes_list = get_plain_list_from_one_of_array(definition)
-        return (
-            plain_datatypes_list == data_type.TIMESTAMP_WITHOUT_TIMEZONE_TYPE
-            or data_type.TIMESTAMP_WITHOUT_TIMEZONE_TYPE in plain_datatypes_list
+    if data_type.ONE_OF_VAR_NAME in definition:
```

Comment:
should all the `is_whatever_type` methods have this oneof handling?

maybe worth doing something like this, to reduce the amount of copypasta (I didn't test this at all, hopefully it's clear enough. Also maybe would be more readable using `lambda` syntax in `get_reftype_function`, idk)
```python
def is_type_included(definition: dict, is_type: Callable[[dict], bool]) -> bool:
    if data_type.ONE_OF_VAR_NAME in definition:
        return bool(
            any(
                is_type(option)
                for option in definition[data_type.ONE_OF_VAR_NAME]
            )
        )
    else:
        return is_type(definition)

def get_reftype_function(type: string) -> Callable[[dict], bool]:
    def is_reftype(definition: dict) -> bool:
        return data_type.REF_TYPE_VAR_NAME in definition and type == definition[data_type.REF_TYPE_VAR_NAME]
    return is_reftype

def is_binary_datatype(definition: dict) -> bool:
    return is_type_included(definition, get_reftype_function(data_type.BINARY_DATA_TYPE))

def is_datetime_with_timezone(definition: dict) -> bool:
    return is_type_included(definition, get_reftype_function(data_type. TIMESTAMP_WITHOUT_TIMEZONE_TYPE))

# etc
```

Reply:
I've seen these methods are also called in other parts of code. So even a simple change may cause unexpected side effects and I've seen it many times during my work on this PR. So since we may have one not only in ref types I really afraid to touch it :)

Reply:
I think this code change is transparent though? I.e. it keeps the same method signature, just changes the implementation to be consistent across methods. So other than the methods that take both `definition` and `property_type`, it shouldn't matter?

mostly just want to doublecheck whether the `oneOf` handling is intentionally excluded from some methods.

Reply:
@Ohh...got your point about Callable. Updated. Also updated methods to get rid of unused property_types. Thanks